### PR TITLE
Turning PDF and HTML content into Markdown

### PR DESCRIPTION
**Pull request by Farid-Karimli**
- Added LlamaParser with GPT4o for parsing PDF documents into Markdown
- Added HTML into Markdown parsing

### CODE REVIEW

**Comment by XThomasBU** [comment]
@Farid-Karimli Could you make this PR into the `dev_branch`

File: code/config.yml
Comment (trgardos) [inline code/config.yml:1]: Try this with `expand_urls: True` to make sure it is loading and parsing the HTML and PDFs from the web site.

**Review by trgardos — CHANGES_REQUESTED** [review]
Needs to be able to pull content from the website without having to manually add files to the data directory.

File: requirements.txt
Comment (trgardos) [inline requirements.txt:1]: I find it brittle to enumerate every package from pip freeze. I'll propose a small set of parent packages.

**Comment by trgardos** [comment]
@Farid-Karimli -- did you merge updates from dev_branch into text_extract?

File: requirements.txt
Comment (XThomasBU) [reply requirements.txt:1]: I used pigar (https://github.com/damnever/pigar) to do this in the dev_branch (https://github.com/DL4DS/dl4ds_tutor/blob/dev_branch/requirements.txt)

Comment (Farid-Karimli) [reply requirements.txt:1]: pigar gave me SEG faults

**Comment by Farid-Karimli** [comment]
I didn't merge the updates from dev_branch into text_extraction

**Comment by Farid-Karimli** [comment]
@XThomasBU @trgardos I refactored the code to parse PDF text and structure without GPT4o, so no image and not the best latex understanding. It also loads PDFs from their URLs temporarily onto the system for parsing.

**Review by trgardos — APPROVED** [review]

**Pull request merged**